### PR TITLE
[WIP] More UI integrations with topnotes

### PR DESCRIPTION
**Pull request by iest**
# Breaking changes
- Change RadioGroup tabbed variant to be circular
- Change RadioGroup API to "giant"
- Remove clear btn type
- Move ticks above Slider
- Some icons renamed:
  - drip -> anDrip
  - dripper -> drip

# New

- Add Title component
- Add textLight colour
- Add table, table-cell, width modifiers
- Add split class to Panel css
- Add pods colour
- Add Pod icon
- Add plus icon
- Add padding zero modifiers
- Add FormInput variant for placeholder & label
- Add Field & FieldGroup
- Add extra-small modifiers
- Add Danger+Clean Btn variant
- Add cover styles
- Add `zeropadding` bool on Panel
- Add Strikethrough modifier
- Lots of new modifiers

# Fixes

- Fix tests
- Fix forminput margins
- Fix anDrip CSS
- Fix Cafetiere spelling mistake
- Fix FormInput to make sure only active when value exists
- Fix Radio; renders children if they exist

# Other changes

- Change Toggle to use value API
- Change slider to be more in line with recent design
- Change FormInput to not sync state with props
- Change font-size variables to rems
- Change Btn warning to danger
- Change Btn CTA padding
- Change black
- Change `--color-text` from grey to black
- Change to `borderless` bool on FormInput — more adaptable
- Use react-motion instead of CSS for animation
- Update HMR & add readme note
- Update babelrc to new react-transform spec
- Stop ragged FormInput labels
- Stop pointer events on slider ticks
- Remove unneeded `root` class from Styleguide
- Remove Overlays styleguide route until we refactor them all
- Rebuild pod icon to be stroked
- Programmatic active state on buttons & datepicker changes as per ui
- Only remove label padding on borderless FormInput
- Move Radio & RadioGroup to use value API
- Make modifiers.css formatting consistent
- Lighten Radio border
- Increase size of slider handle
- Improve Slider style
- Improve cross and pause icons
- Forms.css -> Form.css
- Fatten radio inner
- Correctly pass in the id to FormInput
- Clean up Toggle
- Clean up styles of radio

### CODE REVIEW

File: components/FormInput/FormInput.js [hunk 1]
```diff
@@ -12,13 +13,14 @@ export default class FormInput extends Component {
...
       focus: false,
+      id: uniqueId('forminput'),
     };
   }
 
   componentWillReceiveProps(nextProps) {
     if (!this.state.focus) {
       this.setState({
-        active: nextProps.value.length,
+        active: nextProps.value && nextProps.value.length,
```
Comment (rdjpalmer) [inline components/FormInput/FormInput.js]: Smells like we're keeping state in sync with props. Should be calculated on the fly in `render`.

File: components/Slider/Slider.js [hunk 2]
```diff
@@ -42,6 +36,12 @@ export default class Slider {
           </div>
           : null
         }
+        <ReactSlider withBars
```
Comment (rdjpalmer) [inline components/Slider/Slider.js]: Didn't realise we're using an alpha component?!

File: components/Toggle/Toggle.js [hunk 3]
```diff
@@ -3,22 +3,16 @@ import React, {Component, PropTypes} from 'react';
...
-    super(props);
-    this.handleChange = this.handleChange.bind(this);
-  }
-  handleChange() {
-    this.props.onChange(!this.props.value);
-  }
   render() {
-    const isOn = this.props.value === true;
+    const {value, onChange, className} = this.props;
+    const isOn = value === true;
```
Comment (rdjpalmer) [inline components/Toggle/Toggle.js]: Why do `isOn` at all if it's only used once. Move this logic onto the label's className.

File: globals/modifiers.css [hunk 4]
```diff
@@ -121,3 +147,19 @@
 
 /* Grid helpers */
 .cf {lost-utility: clearfix;}
+
+/* Widths */
+.w1 {  width: 1%;}
```
Comment (rdjpalmer) [inline globals/modifiers.css]: Inconsistant formatting. Also, `width: 1%`... what's the use case?

File: styleguide/styleguide.css [hunk 5]
```diff
@@ -10,12 +10,14 @@
...
 }
 
-.root {
+body {
   background: var(--color-white-dark);
   font: 1.125em/1.5 var(--font-brandon);
   color: var(--color-text);
 }
 
+.root {}
```
Comment (rdjpalmer) [inline styleguide/styleguide.css]: Empty `.root` class is pointless.

File: components/Slider/Slider.js (hunk 2 above)
Comment (iest) [inline components/Slider/Slider.js]: It's only alpha as it's been called that by the maintainer. Works fine as far as I can tell?

File: globals/modifiers.css (hunk 4 above)
Comment (iest) [inline globals/modifiers.css]: For two items in a table cell layout; if you want an item (like an input) to fill as much space as possible, you must give the other item (like a label) an explicit width. `1%` will make the label as large as it needs to be given it's contents, and the input will fill the rest of the space.

File: components/FormInput/FormInput.js [hunk 6]
```diff
@@ -10,57 +11,65 @@ export default class FormInput extends Component {
     this.handleBlur = this.handleBlur.bind(this);
 
     this.state = {
-      active: !!props.value,
       focus: false,
+      id: uniqueId('forminput'),
```
Comment (rdjpalmer) [inline components/FormInput/FormInput.js]: Similar to `active`, this can be done in render as far as I can tell.

Comment (iest) [inline components/FormInput/FormInput.js]: We don't want to have a brand new unique ID on every render call, do we?

File: components/Slider/Slider.js (hunk 2 above)
Comment (rdjpalmer) [inline components/Slider/Slider.js]: Haven't used it myself. Just wanted to make sure we know where we're at with it.

File: components/FormInput/FormInput.js (hunk 6 above)
Comment (rdjpalmer) [inline components/FormInput/FormInput.js]: Lol, yes. Silly me. I have seen two different patterns for this however. One as state, [another as a property on the class](https://github.com/PactCoffee/loggins/blob/feature/my-ui/components/RadioGroup/Radio.js). Let's pick one and stick to it.

File: globals/modifiers.css (hunk 4 above)
Comment (rdjpalmer) [inline globals/modifiers.css]: Very nice :)

File: components/Slider/Slider.js (hunk 2 above)
Comment (iest) [inline components/Slider/Slider.js]: There's [this one](https://github.com/seatgeek/react-slider) which has tests but is deprecated.
There's [this one](https://github.com/xeodou/react-range-slider) which has no tests (just like the one we're using).

Or there's [this one](https://github.com/whoisandie/react-rangeslider) which is tested and is 1.0 but doesn't seem to work at all on my iPhone.

Or we can make our own :)

Comment (rdjpalmer) [inline components/Slider/Slider.js]: We can roll with this one as it seems to work. Change it up later if theres an issue.

**Comment by rdjpalmer** [comment]
Consistent ID usage, and we're good to go.

**Pull request merged**